### PR TITLE
BTRX-609 - ORSP: Convert Tabs into one React Component

### PR DESCRIPTION
**Pull request by vvicario**
## Addresses
https://broadinstitute.atlassian.net/browse/BTRX-625
https://broadinstitute.atlassian.net/browse/BTRX-608
https://broadinstitute.atlassian.net/browse/BTRX-610
https://broadinstitute.atlassian.net/browse/BTRX-611
https://broadinstitute.atlassian.net/browse/BTRX-612
https://broadinstitute.atlassian.net/browse/BTRX-613
https://broadinstitute.atlassian.net/browse/BTRX-609
https://broadinstitute.atlassian.net/browse/BTRX-695
## Changes
Create a react main component that contains grails and react pages for project

## Testing
Describe for the reviewer how to test the specifics of this PR, both positive and negative cases.

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy [hunk 1]
```diff
@@ -7,6 +7,20 @@ import org.broadinstitute.orsp.utils.IssueUtils
 @Slf4j
 class SubmissionController extends AuthenticatedController {
 
+    def show() {
+        Issue issue = queryService.findByKey(params.id)
+        if (issueIsForbidden(issue)) {
+            redirect(controller: 'Index', action: 'index')
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/SubmissionController.groovy]: Is there no user message when this happens? This seems like a poor user experience to follow a link only to not go where you tried to with no explanation as to why. I suggest adding a story for this as it really is unrelated to the larger story of integrating the various react component pages.

File: grails-app/services/org/broadinstitute/orsp/QueryService.groovy [hunk 2]
```diff
@@ -208,7 +208,9 @@ class QueryService implements Status {
 
         // Order is important, see `fundingReport.gsp` for how the UI will represent this data.
         def data = fundings.collect { funding ->
-            String url = applicationTagLib.createLink([controller: funding.issue.controller, action: 'show', absolute: true]) + "/" + funding.issue.projectKey
+            String url = funding.issue.controller == IssueType.CONSENT_GROUP.name ?
+                    applicationTagLib.createLink([controller: funding.issue.controller, action: 'show', absolute: true]) + "/" + funding.issue.projectKey :
+                    applicationTagLib.createLink([controller: "project", action: 'main', absolute: true]) + "?projectKey=" + funding.issue.projectKey
```
Comment (rushtong) [inline grails-app/services/org/broadinstitute/orsp/QueryService.groovy]: nit: move to function

File: grails-app/views/projectDocument/_index.gsp [hunk 3]
```diff
@@ -1,19 +1,12 @@
...
-    };
+<!DOCTYPE html>
+<html>
+<head>
+  <meta name="layout" content="main_component">
+  <title>Project</title>
+</head>
+<body
+  <div id="projectDocument"></div>
+  <asset:javascript src="build/projectDocument.js"/>
```
Comment (rushtong) [inline grails-app/views/projectDocument/_index.gsp]: Beautiful!!!!

File: grails-app/views/projectReview/_adminOnly.gsp [hunk 4]
```diff
@@ -1,13 +0,0 @@
...
-        projectUrl: "${createLink(controller: 'project', action: 'getProject')}",
-        isAdmin: "${session.isAdmin}",
-        loadingImage: "${resource(dir: 'images', file: 'loading-indicator.svg')}",
-        userSessionUrl: "${createLink(controller: 'authenticated', action: 'getSessionUser')}",
-        updateAdminOnlyPropsUrl: "${createLink(controller: 'project', action: 'updateAdminOnlyProps')}"
-    };
-</g:javascript>
-
-<div id="adminOnly"></div>
-<asset:javascript src="build/adminOnly.js"/>
```
Comment (rushtong) [inline grails-app/views/projectReview/_adminOnly.gsp]: Love it!

File: src/main/webapp/consentGroup/NewConsentGroup.js [hunk 5]
```diff
@@ -114,7 +114,7 @@ class NewConsentGroup extends Component {
         .then(resp => {
           // TODO: window.location.href is a temporal way to redirect the user to project's consent-group page tab. We need to change this after
           // transitioning from old gsps style is solved.
-          window.location.href = [this.props.serverURL, projectType, "show", this.props.projectKey, "?tab=consent-groups&new"].join("/");
+          window.location.href = [this.props.serverURL, "project", "main?projectKey=" + this.props.projectKey + "&tab=consent-groups&new"].join("/");
```
Comment (rushtong) [inline src/main/webapp/consentGroup/NewConsentGroup.js]: I'm a little sad that we still need `window.location.href`, but I understand it. Hopefully we can get away from this when more of the app is ported over to react.

File: src/main/webapp/projectContainer/Comments.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+export const Comments = hh(class Comments extends Component {
+
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return (
+      div({dangerouslySetInnerHTML: { __html: this.props.commentsContent } },[])
```
Comment (rushtong) [inline src/main/webapp/projectContainer/Comments.js]: Nice!

File: src/main/webapp/projectContainer/ConsentGroups.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        minWidth: 1000,
+        minHeight: 500,
+        closeOnEscape: true,
+        hide: { effect: "fadeOut", duration: 300 },
+        show: { effect: "fadeIn", duration: 300 },
+        dialogClass: "no-titlebar"
+      }).parent().removeClass("ui-widget-content");
+      $(".ui-dialog-titlebar").hide();
+    });
+    // Display for 8 seconds a message indicating the submission of a new consent group. This is temporary until this page is moved to react.
```
Comment (rushtong) [inline src/main/webapp/projectContainer/ConsentGroups.js]: nit: Can you add a link to the story for this TODO?

File: grails-app/controllers/org/broadinstitute/orsp/AuthenticatedController.groovy [hunk 8]
```diff
@@ -150,14 +150,18 @@ class AuthenticatedController implements Interceptor, UserInfo {
             if (comment == null) {
                 flash.error = "Error saving comment"
                 log.error("Error saving comment for issue '" + params.id + "': null")
-                redirect([action: "show", controller: issue.controller, params: [id: params.id, tab: "comments"]])
+                issue.type == IssueType.CONSENT_GROUP.name ?
+                redirect([action: "show", controller: issue.controller, params: [id: params.id, tab: "comments"]]) :
```
[the remark this answers is not in the document]
Comment (tadeork) [reply grails-app/controllers/org/broadinstitute/orsp/AuthenticatedController.groovy]: I think it's better to have an if statement here.

File: grails-app/controllers/org/broadinstitute/orsp/ConsentGroupController.groovy [hunk 9]
```diff
@@ -361,7 +361,10 @@ class ConsentGroupController extends AuthenticatedController {
         } catch (Exception e) {
             flash.error = "Unable to attach consent document: " + e.getMessage()
         }
-        redirect(controller: issue.controller, action: "show", params: [id: issue.projectKey, tab: "consent-groups"])
+        issue.type == IssueType.CONSENT_GROUP.name ?
+                redirect(controller: issue.controller, action: "show", params: [id: issue.projectKey, tab: "consent-groups"]) :
+                redirect(controller: "project", action: "main", params: [projectKey: issue.projectKey, tab: "consent-groups"])
+
```
[the remark this answers is not in the document]
Comment (vvicario) [reply grails-app/controllers/org/broadinstitute/orsp/ConsentGroupController.groovy]: :+1:

**Pull request merged**